docs/vo_service_discovery/vo_service_discovery.rst

```diff
@@ -0,0 +1,202 @@
+.. doctest-skip-all
+
+.. _astroquery.vo_service_discovery:
+
+*************************************************************************
+Virtual Observatory Service Discovery (`astroquery.vo_service_discovery`)
+*************************************************************************
+
+Getting Started
+===============
+
+Archives from around the world have registered their services with the NASA
+Astronomy Virtual Observatory (NAVO) directory. A web interface is available
+at http://vao.stsci.edu/keyword-search/. This module provides methods
+to query this directory by service type, keywords, wavelength band or coverage,
+and source or publisher.
+
+Here is an example showing how to find information about conesearch services
+provided by the NASA Extragalactic Database (NED).
+
+.. code-block:: python
+
+    >>> from astroquery.vo_service_discovery import Registry
+    >>> results = Registry.query(source='ned', service_type='cone')
+    >>> print(results)
+
+                                waveband                         ... service_type
+        -------------------------------------------------------- ... ------------
+                                                        infrared ...   conesearch
+        radio#millimeter#infrared#optical#uv#euv#x-ray#gamma-ray ...   conesearch
+
+The result of the query is an `astropy.table.Table` object. The methods of
+`~astropy.table.Table`
+can be used to inspect the search result.
+
+.. code-block:: python
+
+    >>> print(results.colnames)
+
+        ['waveband', 'short_name', 'ivoid', 'res_description', 'access_url',
+         'reference_url', 'publisher', 'service_type']
+
+The value of ``'access_url'`` specifies the endpoint for the service.
+
+Documentation of the service may be found in the ``'reference_url'`` values.
+You can use a function from the Python standard library to open the second url
+in a web browser.
+
+.. code-block:: python
+
+    >>> import webbrowser
+    >>> webbrowser.open(results['reference_url'][1].decode())
```

Comment:
This is basically a safety measure as not all URLs across all registries are stored decoded. While it would be best if the user did not have to decode any results, there is little explicit documentation within the IVOA standards about whether all the URLs should be escaped or not, and some inconsistency remains among the ~20k current resources.

Reply:
IMHO such technicalities should be hidden from users if possible. But if it is impossible, perhaps an explanation in the doc on why decode is needed here?

Reply:
The commit comment for the decode() says it's needed because the reference_url value is a byte-string.  I think that's an issue with the VOTable parser in Astropy, but a little more looking is needed.  

If the reference_url were a str, it may not need decoding regardless of whether the value were already URL-encoded.

https://github.com/astropy/astroquery/pull/1158#discussion_r240796583

Reply:
Apologies, I should have looked closer. There have been other places where we safety-decode URLs in strings from the registry and hide it from the user, I think Tom's right that this is a separate thing.

Reply:
I won't be surprised if this stemmed from `io.votable`, as that package needs some maintenance. 😅

Reply:
Where did we end up coming down on this?  I really don't like the idea of telling users to use `decode`.  I'd slightly prefer a hack-fix to the votable parser within this module over documentation telling users to change string types.